torchgeo/datasets/geo.py

```diff
@@ -359,6 +360,11 @@ class RasterDataset(GeoDataset):
     #: Color map for the dataset, used for plotting
     cmap: ClassVar[dict[int, tuple[int, int, int, int]]] = {}
 
+    #: Nodata value used in the raster for pixels outside the satellite acquisition
+    #: Optionally set this if the raster lack nodata-mask and nodata value
+    #: and you want need it to override `RasterDataset._footprint_from_datasource()`
+    nodata_value: float | None = None
```

Comment:
I left this in here. _if_ someone wants to use this in combination with overriding `_footprint_from_datasource` to call `datasets.utils.get_valid_footprint_from_datasource`, this is passed through for them.